File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        var map = new Dictionary<int, int>();
        foreach (var num in nums) { // O(n)
            if (map.ContainsKey(num)) {
                map[num]++;
            } else {
                map.Add(num, 1);
            }
        }

        var buckets = new List<int>[nums.Length + 1];
        foreach (var kvp in map) {
            if (buckets[kvp.Value] is null) {
                buckets[kvp.Value] = new List<int>();
            }
            buckets[kvp.Value].Add(kvp.Key);
        }

        var result = new List<int>();
        for (int i = buckets.Length - 1; i >= 0 && result.Count < k; i--){
            if (buckets[i] is not null) result.AddRange(buckets[i]);
        }

        return result.ToArray();
    }
}

File: Data Structures & Algorithms/sort-colors/submission-1.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        std::array<int, 3> counts{};

        for (const int& num : nums) {
            counts[num]++;
        }

        int i = 0;
        for (int j = 0; j < 3; j++) {
            while (counts[j] > 0) {
                nums[i++] = j;
                counts[j]--; 
            }
        } 
    }
};